src/app/preferences/general-preferences/general-preferences.controller.ts

```diff
@@ -1,18 +1,12 @@
-/// <reference path="general-preferences.service.ts"/>
-
 namespace Core {
-  export function GeneralPreferencesController($scope, generalPreferencesService: GeneralPreferencesService) {
+  export function GeneralPreferencesController($scope, $window: ng.IWindowService) {
```

Comment:
Why did you delete the GeneralPreferencesService?   Understand the responsibilities of MVC model, where Model = service, View = html, and Controller = controller here. In MVC, only models (= services) should do any meaningful stuffs, such as persisting to local storage.

Reply:
done